readthedocs/builds/version_slug.py

```diff
@@ -94,18 +79,23 @@ def _normalize(self, content):
         """
         return re.sub('[/%!?]', '-', content)
 
-    def slugify(self, content):
+    def is_valid(self, content):
+        return self.test_pattern.match(content)
+
+    def slugify(self, content, check_pattern=True):
```

Comment:
Why did you add the `check_pattern` if we are using it always in `False`? It didn't exist before and now it is changing the default behavior to raise an exception.

Reply:
So, I added it bc it was weird to don't do anything when the slug wasn't right, let me see if I can change the other code to return None for an invalid slug